bigbone/src/test/kotlin/social/bigbone/api/method/AnnouncementMethodsTest.kt

```diff
@@ -0,0 +1,101 @@
+package social.bigbone.api.method
+
+import io.mockk.verify
+import org.amshove.kluent.AnyException
+import org.amshove.kluent.invoking
+import org.amshove.kluent.shouldBeEqualTo
+import org.amshove.kluent.shouldNotThrow
+import org.amshove.kluent.shouldThrow
+import org.junit.jupiter.api.Test
+import social.bigbone.MastodonClient
+import social.bigbone.Parameters
+import social.bigbone.api.exception.BigBoneRequestException
+import social.bigbone.testtool.MockClient
+
+class AnnouncementMethodsTest {
+    @Test
+    fun `Get all read announcements`() {
+        val client = MockClient.mock("announcements.json")
+        val announcementMethods = AnnouncementMethods(client)
+        val announcements = announcementMethods.getAllAnnouncements(withDismissed = true).execute()
+
+        announcements.size shouldBeEqualTo 3
+
+        verify {
+            client.get(
+                path = "/api/v1/announcements",
+                query = any<Parameters>()
+            )
+        }
+    }
+
+    @Test
+    fun `Dismiss an announcement and verify that the method was called correctly`() {
+        val client = MockClient.mock("announcements.json")
+        val announcementMethods = AnnouncementMethods(client)
+
+        invoking { announcementMethods.dismissAnnouncement("8") } shouldNotThrow AnyException
```

Comment:
That’s a bit verbose, but works. Unless explicitly set up that way, unit tests don’t expect exceptions anyway, so the test would fail anyway.

```suggestion
        announcementMethods.dismissAnnouncement("8")
```